codelingo/lexiconsdk
# TS/JS Lexicon stub

First pass at a lexicon that supports:
* JavaScript
* TypeScript
* JSX/TSX

I gave up on the TypeScript parser and have used the [`@babel/parser`](https://babeljs.io/docs/en/next/babel-parser.html) library instead (which [I discovered here](https://astexplorer.net/).)

To test it (requires node):
1. Clone the repo
2. `npm install` for packages
3. `tsc` to build (part of the `typescript` package -- hopefully works, you may need to do `npm install --global typescript`)
4. Run it against a JavaScript/TypeScript codebase:
```
node dist/main.js parse-project 0_0 ~/dev/something
```

Notes:
* It's based on the `csharp` example, and what I could glean from others.
* Everything outputs to the `ts` namespace (this could be changed).
* It should have pretty good coverage of JS, JSX, TS and TSX formats, except for bleeding-edge stuff, which I've left out for now.
* I didn't really understand `otherSiblings`, so that has not been implemented yet.
* The `makeRelativePath()` function is probably too simplistic. It's not doing as much as the `csharp` example was doing.
* None of the Dockerfile work has been done.
* No tests yet.
* It doesn't implement the `replace-properties` action yet.
